### DISCUSSION TITLE
totals etc. when filtering data

### DISCUSSION THREAD

**Discussion by Skeeve** [alpinejs/alpine#2402] [3. Feature ideas]
I asked a question on [stackoverflow](https://stackoverflow.com/questions/70076880/count-elements-in-alpine-js) about how to get sums or, for that matter, totals, when filtering some data.

Finally I came to the conclusion that it is currently impossible.

In the example, the individualcount does work, as the template's root element has an x-data which gets initialized.

But it seems to be impossible to set on the template, some initialization for data a level above. Much like the first part of a traditional `for (;;)` loop.

As I'm very fresh to alpinejs, I just read about it two days ago, I might miss something obvious. But if not: Maybe it's an idea to have some kind of initialization where we have an `x-for`?

**Comment by SimoTod** [comment]
Not really, but you don't need it.
first of all, the way you use effect is not the one it was meant to.
`x-effect` runs at the beginning and every time a variable mentioned in it changes outside of another effect while you are using it as it was x-init.
Because you don't have control about how many time runs (and you don't care either, Alpine is meant to be more declarative than imperative), each execution should return the same result if the search string hasn't changed.
Incrementing a counter as a side effect is not accurate, because it will only run for new elements added to the filtered array.

A different approach which would achieve what you need would be this one: https://codepen.io/SimoTod/pen/rNzEJRW

**Reply by Skeeve** [reply]
Thanks a lot. That's almost what I already did in the meantime, but it gave me some more ideas how to improve.
